packages/core/src/Stores/notification-store.js

```diff
@@ -891,7 +891,31 @@ export default class NotificationStore extends BaseStore {
                                   text: localize('Yes, increase my limits'),
                               }
                             : {
-                                  route: routes.cashier_p2p_profile,
+                                  onClick: () => {
+                                      // TODO: Reroute this to my profile properly during p2p routing refactor
+                                      const routePromise = new Promise(resolve => {
+                                          // Need setTimeout as routing to p2p has a delay on mobile
+                                          if (window.location.pathname.includes(routes.cashier)) {
+                                              this.root_store.common.routeTo(routes.cashier_p2p);
+                                              setTimeout(
+                                                  () => {
+                                                      if ('redirectTo' in this.p2p_redirect_to) resolve();
+                                                  },
+                                                  isMobile() ? 1500 : 0
```

Comment:
on desktop, the time to navigate between pages is significantly faster in comparison to mobile, hence why the time is set to 0. initially I had no setTimeout and it worked fine, but for mobile I needed the setTimeout